decomposed/extensions/MDDE_LDM/Profile/Model/Methods/mdde_InitializeExportExtension.xml

```diff
@@ -0,0 +1,99 @@
+   Dim ent
+   For Each ent in objPackage.Entities
+      If Not ent.IsShortCut Then
+         applyTemplateOnObject objModel, ent, &quot;mddex_Entity_Export_Template&quot;, &quot;mddex_Entity_Export_Content&quot;
+         applyTemplateOnObject objModel, ent, attributeTemplateName, attributeExtendedAttributeName            
+      End If   
+   Next
+   
+   &#39;Loop over EntityAttributes
+   Dim entAttr
+   For Each entAttr in objPackage.EntityAttributes
+      If Not entAttr.IsShortCut Then
+         applyTemplateOnObject objModel, entAttr, &quot;mddex_EntityAttribute_Export_Template&quot;, &quot;mddex_EntityAttribute_Export_Content&quot;
+         applyTemplateOnObject objModel, entAttr, attributeTemplateName, attributeExtendedAttributeName            
+      End If   
+   Next
+   
+   &#39;Loop over Mappings
+   Dim map
+   For Each map in objPackage.Mappings
+       If Not map.IsShortCut Then
+         applyTemplateOnObject objModel, map, &quot;mddex_BaseClassifierMapping_Export_Template&quot;, &quot;mddex_BaseClassifierMapping_Export_Content&quot;
+         applyTemplateOnObject objModel, map, attributeTemplateName, attributeExtendedAttributeName            
+      End If   
+   Next
+   
+   &#39;Loop over AttributeMappings
+   Dim attrMap
+   For Each attrMap in objPackage.ModelEntityAttributeMappings
+      If Not attrMap.IsShortCut Then
+         applyTemplateOnObject objModel, attrMap, &quot;mddex_BaseStructuralFeatureMapping_Export_Template&quot;, &quot;mddex_BaseStructuralFeatureMapping_Export_Content&quot;         
+      End If   
+   Next
+   
+   &#39;Loop over packages
+   Dim childPackage
+   For Each childPackage in objPackage.Packages
+      processPackage objModel, childPackage, attributeTemplateName, attributeExtendedAttributeName
+   Next
+   
+End Sub
+
+&#39;Apply template on object and check for translation errors
+Sub applyTemplateOnObject(objModel, objItem, templateName, extendedAttributeName)
+   &#39;Process the object specific template
+   Dim evaluatedTemplate: evaluatedTemplate = objItem.EvaluateTemplateFor(templateName, cMDDE_EXPORT_EXTENSION_CODE)
+   &#39;Clear the extended attribute if it is set
+   If Not objItem.getExtendedAttributeText(extendedAttributeName) = &quot;&quot; Then
+      objItem.setExtendedAttribute extendedAttributeName, Nothing
+   End If
+   
+   &#39; Only set the extended attribute if the template evaluation gave a non-empty result
+   If Not evaluatedTemplate = empty And Not evaluatedTemplate = &quot;&quot; Then
+      objItem.setExtendedAttribute extendedAttributeName, evaluatedTemplate     
+      If GTLOutputHasErrors(objItem.getExtendedAttribute(extendedAttributeName)) Then
+         WriteError (&quot;Error parsing template &quot; &amp; templateName &amp; &quot; for &quot; &amp; objItem.Code)
+         &#39;Set success indicator to false
+         objModel.setExtendedAttribute cMDDE_EXT_CODE &amp; &quot;.&quot; &amp; cMDDE_EXPORT_EXTENSION_INITIALIZATION_INDICATOR, False
```

Comment:
Maybe good to exit function here so we don't continue?

Reply:
Changed errorhandling with explicit messagebox in case something fails